src/library/searchqueryparser.cpp

```diff
@@ -115,39 +120,54 @@ void SearchQueryParser::parseTokens(QStringList tokens,
             continue;
         }
 
+        bool exact = m_exactTextMatcher.indexIn(token) != -1;
         bool negate = token.startsWith(kNegatePrefix);
         std::unique_ptr<QueryNode> pNode;
 
         if (m_fuzzyMatcher.indexIn(token) != -1) {
             // TODO(XXX): implement this feature.
         } else if (m_textFilterMatcher.indexIn(token) != -1) {
-            QString field = m_textFilterMatcher.cap(1);
-            QString argument = getTextArgument(
-                m_textFilterMatcher.cap(2), &tokens).trimmed();
+            QString field = m_textFilterMatcher.cap(1);            
+            QString argument; 
+            if (exact) {
+                argument = getTextArgument(
+                            m_exactTextMatcher.cap(2), &tokens).trimmed();
+            } else {
+                argument = getTextArgument(
+                            m_textFilterMatcher.cap(2), &tokens).trimmed();
+            }
 
             if (!argument.isEmpty()) {
                 if (field == "crate") {
                     pNode = std::make_unique<CrateFilterNode>(
-                          &m_pTrackCollection->crates(), argument);
+                            &m_pTrackCollection->crates(), argument);
+                } else if (exact) {
+                    pNode = std::make_unique<ExactFilterNode>(
+                            m_pTrackCollection->database(),
+                            m_fieldToSqlColumns[field], argument);
                 } else {
                     pNode = std::make_unique<TextFilterNode>(
-                          m_pTrackCollection->database(), m_fieldToSqlColumns[field], argument);
+                            m_pTrackCollection->database(),
+                            m_fieldToSqlColumns[field], argument);
                 }
+            } else {
+                pNode = std::make_unique<SqlNode>(
+                        field + " IS NULL");
```

Comment:
@jmigual I don't know why it says I posted this 22 hours ago. This is from 2 weeks ago.

[https://github.com/mixxxdj/mixxx/pull/1117#pullrequestreview-42580059] this is what I commented 22 hours ago.